src/main/kotlin/ai/blindspot/ktoolz/extensions/OtherExtensions.kt

```diff
@@ -87,3 +87,70 @@ inline fun <T : Any> T.applyIf(shouldApply: Boolean, block: T.() -> Unit): T {
     return this
 }
 
+/**
+ * Shortens the string if needed. For example, returns "ABCD…" when called "ABCDEFHG.(5, "…")
+ */
+fun String.restrictLengthWithEllipsis(maxLength: Int, ellipsis: String = "…"): String {
+    if (this.length <= maxLength) return this
+    return this.substring(0, maxLength - ellipsis.length) + ellipsis
+}
+
+/**
+ * Creates a string like "className(details)", for example "Double(42.0)"
+ * Useful for implementing .toString() method (but in such case the first parameter MUST be passed).
+ */
+fun Any.toLongDebugString(description: String? = null, brackets: String = "()", className: String? = null): String {
+    val actualDescription = description ?: this.toString()
+    val actualClassName = className ?: this.javaClass.simpleName
+    val actualBrackets = if (brackets.length == 2) brackets else "<>"
+    return "$actualClassName${actualBrackets[0]}$actualDescription${actualBrackets[1]}"
+}
+
+/**
+ * Extracts the essential information from an object (most usually a string), whose toString() call result
+ * has a similar format as the output of function [toLongDebugString] .
+ * If the format is not similar, it simply returns this.toString() .
+ *
+ * For example, returns "42.0" for string "Double(42.0)".
+ * Returns "John 42, Peter 31" for string "EmployeeList{John 42, Peter 31}"
+ *
+ * Typical usage: the toString() method of a large collection-like object should contain only a very short description
+ *  of each collection item. This is done, for example, in [itemsToDebugString] ().
+ */
+fun Any?.toShortDebugString(): String {
+    val longString = this.toString()
+    if (longString.isEmpty()) return "EMPTY STRING" // Should not happen
+    val startsWithLetter = longString[0].let { it in 'a'..'z' || it in 'A'..'Z' }
+    if (!startsWithLetter) return longString // The format does not resemble an output of [toLongDebugString]
+
+    val bracketPairs = setOf("()", "[]", "<>", "{}")
+    for (pair in bracketPairs) {
+        if (longString.last() == pair[1]) {
+            val after = longString.substringAfter(pair[0])
+            if (after.length < 2) return longString
+            return after.substring(0, after.length - 1) // Omit the last char
+        }
+    }
+    return longString // The format does not resemble an output of [toLongDebugString]
+}
+
+/**
+ * Formats a collection to a readable string like "3 colors: red, yellow, green".
+ */
+inline fun <T> Collection<T>.itemsToDebugString(
+    itemsType: String = "items", // or "employees" etc.
+    separator: String = ", ", // inserted between (string representations of) collection items
+    itemLength: Int = 30, // max length
+    totalLength: Int = 200,
+    itemToString: (T) -> String = { item -> item.toShortDebugString() }
+): String {
```

Comment:
The default value of itemToString is OK?

Reply:
I would more expect using simple `toString` as default value because that's default behavior of everything, but it is actually up to you, `toShortDebugString` seems suitable as well

Reply:
The default `item -> item.toShortDebugString()` demonstrates both a typical usage of `itemsToDebugString()` and of `toShortDebugString()` function. Otherwise I would surely prefer `.toString` default ; this way it's hard to decide...